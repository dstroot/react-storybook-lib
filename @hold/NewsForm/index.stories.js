import React from "react";
import { storiesOf } from "@storybook/react";

// components
import NewsForm from "../NewsForm";

// NewsForm section
storiesOf("NewsForm", module).add("show NewsForm", () => <NewsForm />);
