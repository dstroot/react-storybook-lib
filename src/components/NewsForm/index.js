import React from "react";
import Button from "../Button";
import { Link } from "@reach/router";

const NewsForm = ({
  valid,
  submitted,
  success,
  message,
  draft,
  datetime,
  headline,
  cutline,
  story,
  category,
  image,
  onSubmit,
  onChange
}) => {
  return (
    <>
      <h3 className="mt-3 font-weight-light">News Story:</h3>
      <form
        onSubmit={onSubmit}
        noValidate
        className={!valid ? "was-validated" : ""}
      >
        <fieldset disabled={submitted} aria-busy={submitted}>
          <div className="form-row align-items-center">
            {/* Date */}
            <div className="form-group col-md-6">
              <label htmlFor="date">Date</label>
              <input
                name="datetime"
                type="date" // datetime-local  date
                className="form-control"
                value={datetime}
                onChange={onChange}
                required
              />
              <div className="invalid-feedback">Please enter a date.</div>
              <div className="valid-feedback">Looks good!</div>
            </div>
          </div>

          {/* Draft */}
          <div className="form-group form-check col-md-6">
            <input
              type="checkbox"
              className="form-check-input"
              name="draft"
              checked={draft}
              onChange={onChange}
            />
            <label className="form-check-label" htmlFor="draft">
              Draft
            </label>
          </div>
          {/* end row */}

          {/* Headline */}
          <div className="form-group">
            <label htmlFor="headline">Headline</label>
            <textarea
              name="headline"
              type="text"
              className="form-control"
              placeholder="Please enter a headline"
              value={headline}
              onChange={onChange}
              required
            />
            <div className="invalid-feedback">Please enter a headline.</div>
            <div className="valid-feedback">Looks good!</div>
          </div>

          {/* cutline area */}
          <div className="form-group">
            <label htmlFor="cutline">Cutline</label>
            <textarea
              name="cutline"
              className="form-control"
              placeholder="Please enter a cutline"
              value={cutline}
              onChange={onChange}
              required
            />
            <div className="invalid-feedback">Please enter some text.</div>
            <div className="valid-feedback">Looks good!</div>
          </div>

          {/* story area */}
          <div className="form-group">
            <label htmlFor="story">Story</label>
            <textarea
              name="story"
              className="form-control"
              placeholder="Please enter a story"
              rows={10}
              value={story}
              onChange={onChange}
              required
            />
            <div className="invalid-feedback">Please enter some text.</div>
            <div className="valid-feedback">Looks good!</div>
          </div>

          <div className="form-row align-items-center">
            {/* category */}
            <div className="form-group col-md-6">
              <label htmlFor="category">Category</label>
              <select
                className="form-control"
                name="category"
                value={category}
                onChange={onChange}
              >
                <option category="Company News">Company News</option>
                <option category="Movers and Shakers">
                  Movers and Shakers
                </option>
                <option category="Product News">Product News</option>
              </select>
              <div className="invalid-feedback">Please select one.</div>
              <div className="valid-feedback">Looks good!</div>
            </div>

            {/* image */}
            <div className="form-group col-md-6">
              <label htmlFor="image">image</label>
              <input
                name="image"
                type="text"
                className="form-control"
                placeholder="Please enter the image URL"
                value={image}
                onChange={onChange}
              />
              <div className="invalid-feedback">Please enter the URL.</div>
              <div className="valid-feedback">Looks good!</div>
            </div>
          </div>
          {/* end row */}
        </fieldset>

        {/* Submit button */}
        <div className="form-row justify-content-between mt-3">
          <div className="col-md-5">
            <Link to="/news/edit" className="btn btn-block btn-secondary">
              Cancel
            </Link>
          </div>
          <div className="col-md-5">
            <Button submitted={submitted} success={success} />
          </div>
        </div>
      </form>
      <div className="row">
        <div className="col-md-6 offset-md-3">
          {message ? <p className="text-danger">{message}</p> : ""}
        </div>
      </div>
    </>
  );
};

export default NewsForm;
