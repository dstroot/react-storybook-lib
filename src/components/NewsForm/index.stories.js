import React from "react";
import { storiesOf } from "@storybook/react";

// components
import NewsForm from "../NewsForm";

// NewsForm section
storiesOf("NewsForm", module)
  .addParameters({
    // default for all stories in this book
    info: {
      inline: true,
      header: false
    }
  })
  .add("show NewsForm", () => (
    <NewsForm
      valid={true}
      submitted={false}
      success="yes"
      message="Hello"
      datetime="Dec 25, 2018"
      headline="Headline"
      cutline="cutline"
      story="story"
      category="Company News"
      image="Image"
    />
  ))
  .add("show invalid NewsForm", () => (
    <NewsForm
      valid={false}
      submitted={false}
      success="yes"
      message="Hello"
      datetime="Dec 25, 2018"
      headline="Headline"
      cutline="cutline"
      story="story"
      category="Company News"
      image="Image"
    />
  ));
