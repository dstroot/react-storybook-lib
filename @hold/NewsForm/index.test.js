import React from "react";
import renderer from "react-test-renderer";

import NewsForm from "../NewsForm";

const onSubmit = () => {
  return null;
};

const onChange = () => {
  return null;
};

test("NewsForm renders some content", () => {
  const component = renderer.create(
    <NewsForm
      valid={true}
      submitted={false}
      success="yes"
      message="Hello"
      datetime="Dec 25, 2018"
      headline="Headline"
      cutline="cutline"
      story="story"
      category="Company News"
      image="Image"
      onSubmit={onSubmit}
      onChange={onChange}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
