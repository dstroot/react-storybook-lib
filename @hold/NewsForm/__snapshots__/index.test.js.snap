// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsForm renders some content 1`] = `
Array [
  <h3
    className="mt-3 font-weight-light"
  >
    News Story:
  </h3>,
  <form
    className=""
    noValidate={true}
    onSubmit={[Function]}
  >
    <fieldset
      aria-busy={false}
      disabled={false}
    >
      <div
        className="form-row align-items-center"
      >
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="date"
          >
            Date
          </label>
          <input
            className="form-control"
            name="datetime"
            onChange={[Function]}
            required={true}
            type="date"
            value="Dec 25, 2018"
          />
          <div
            className="invalid-feedback"
          >
            Please enter a date.
          </div>
          <div
            className="valid-feedback"
          >
            Looks good!
          </div>
        </div>
      </div>
      <div
        className="form-group form-check col-md-6"
      >
        <input
          className="form-check-input"
          name="draft"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          className="form-check-label"
          htmlFor="draft"
        >
          Draft
        </label>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="headline"
        >
          Headline
        </label>
        <textarea
          className="form-control"
          name="headline"
          onChange={[Function]}
          placeholder="Please enter a headline"
          required={true}
          type="text"
          value="Headline"
        />
        <div
          className="invalid-feedback"
        >
          Please enter a headline.
        </div>
        <div
          className="valid-feedback"
        >
          Looks good!
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="cutline"
        >
          Cutline
        </label>
        <textarea
          className="form-control"
          name="cutline"
          onChange={[Function]}
          placeholder="Please enter a cutline"
          required={true}
          value="cutline"
        />
        <div
          className="invalid-feedback"
        >
          Please enter some text.
        </div>
        <div
          className="valid-feedback"
        >
          Looks good!
        </div>
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="story"
        >
          Story
        </label>
        <textarea
          className="form-control"
          name="story"
          onChange={[Function]}
          placeholder="Please enter a story"
          required={true}
          rows={10}
          value="story"
        />
        <div
          className="invalid-feedback"
        >
          Please enter some text.
        </div>
        <div
          className="valid-feedback"
        >
          Looks good!
        </div>
      </div>
      <div
        className="form-row align-items-center"
      >
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="category"
          >
            Category
          </label>
          <select
            className="form-control"
            name="category"
            onChange={[Function]}
            value="Company News"
          >
            <option
              category="Company News"
            >
              Company News
            </option>
            <option
              category="Movers and Shakers"
            >
              Movers and Shakers
            </option>
            <option
              category="Product News"
            >
              Product News
            </option>
          </select>
          <div
            className="invalid-feedback"
          >
            Please select one.
          </div>
          <div
            className="valid-feedback"
          >
            Looks good!
          </div>
        </div>
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="image"
          >
            image
          </label>
          <input
            className="form-control"
            name="image"
            onChange={[Function]}
            placeholder="Please enter the image URL"
            type="text"
            value="Image"
          />
          <div
            className="invalid-feedback"
          >
            Please enter the URL.
          </div>
          <div
            className="valid-feedback"
          >
            Looks good!
          </div>
        </div>
      </div>
    </fieldset>
    <div
      className="form-row justify-content-between mt-3"
    >
      <div
        className="col-md-5"
      >
        <a
          className="btn btn-block btn-secondary"
          href="/news/edit"
          onClick={[Function]}
        >
          Cancel
        </a>
      </div>
      <div
        className="col-md-5"
      >
        <button
          className="btn btn-block btn-primary"
          type="submit"
        >
          <i
            className="fa"
          />
           Â  
          Submit
        </button>
      </div>
    </div>
  </form>,
  <div
    className="row"
  >
    <div
      className="col-md-6 offset-md-3"
    >
      <p
        className="text-danger"
      >
        Hello
      </p>
    </div>
  </div>,
]
`;
